### PR TITLE
Initial recipe test for NetworkIt

### PR DESCRIPTION
**Pull request by lucventurini**
Initial recipe for NetworkIt.

@conda-forge/help-python-c
@kit-parco

Checklist

- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [] Package does not vendor other packages
- [X] Build number is 0
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by lucventurini** [comment]
The recipe should be ready, waiting for approval.
Adding @avdgrinten and @angriman to the pull request.

**Comment by lucventurini** [comment]
@bmabey @eklitzke 
Apologies for bumping, but GitHub suggested your names when trying to ping someone on this issue. How would I have someone review the recipe, please?

**Comment by avdgrinten** [comment]
@lucventurini Could you please add me, @angriman and @fabratu as a maintainer to the configuration file? Afterwards, we can ping the conda-forge maintainers to merge this PR.

**Comment by lucventurini** [comment]
Hi @avdgrinten, done!

File: recipes/networkit/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+{% set file_ext = "tar.gz" %}
+{% set hash_type = "sha256" %}
+{% set hash_value = "aefef17589991f6e4a04451164eb2d379c7301c9c85e787ef3ddf2f4777b922d" %}
+
+package:
+  name: '{{ name|lower }}'
+  version: '{{ version }}'
+
+source:
+  - git_url: 'https://github.com/kit-parco/{{name}}.git'
```
Comment (chrisburr) [inline recipes/networkit/meta.yaml]: Can you build from the PyPI source tarball instead? (See the example recipe for how to do this.)

See: http://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos

File: recipes/networkit/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    - scikit-learn >=0.20.3
+    
+test:
+  imports:
+    - networkit
+    - networkit.nkadapters
+
+about:
+  home: https://github.com/kit-parco/networkit
+  # Remember to specify the license variants for BSD, Apache, GPL, and LGLP.
```
Comment (chrisburr) [inline recipes/networkit/meta.yaml]: Can you remove these comments which are from the example

File: recipes/networkit/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    - '{{PYTHON}} -m pip install --ignore-installed -vv .'
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake >=3.5
+    - ninja >=1.9.0
+  host:
+    - python >=3.3
```
Comment (chrisburr) [inline recipes/networkit/meta.yaml]: The Python version should only be constrained in this way when the package is `noarch`
```suggestion
    - python
```
(If not you have one build per Python version)

File: recipes/networkit/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    - cython >=0.21
+    - openmp    
+    - sphinx
+    - sphinx_bootstrap_theme
+    - wheel
+    - scikit-learn >=0.20.3
+    - seaborn >=0.8.1
+    - git
+  run:
+    - python >=3.3
```
Comment (chrisburr) [inline recipes/networkit/meta.yaml]: Similarly
```suggestion
    - python
```

**Comment by lucventurini** [comment]
Dear @chrisburr ,
                              everything seems in order now. Would it be possible to merge into Conda forge?

File: recipes/networkit/meta.yaml (hunk 1 above)
Comment (lucventurini) [reply recipes/networkit/meta.yaml]: Done.

**Comment by lucventurini** [comment]
@conda-forge/help-python

We would kindly ask for reviewing and confirming, so that the package can be merged into conda-forge. Thank you.

File: recipes/networkit/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,75 @@
+{% set name = "networkit" %}
+{% set version = "5.0" %}
+{% set file_ext = "tar.gz" %}
+{% set hash_type = "sha256" %}
```
Comment (xhochy) [inline recipes/networkit/meta.yaml]: Please remove these two lines, the `sha256` is explicitly set below.

File: recipes/networkit/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    - cython >=0.21
+    - openmp    
+    - sphinx
+    - sphinx_bootstrap_theme
+    - wheel >=0.28
+    - scikit-learn >=0.20.3
+    - seaborn >=0.8.1
+    - networkx
+    - tabulate
+    - matplotlib
```
Comment (xhochy) [inline recipes/networkit/meta.yaml]: Please use `matplotlib-base` when you don't explicitly require the Qt frontend.
```suggestion
    - matplotlib-base
```

File: recipes/networkit/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    - ipython
+    - jupyter
+  run:
+    - python
+    - openmp
+    - scikit-learn >=0.20.3
+    - networkx
+    - tabulate
+    - pandas
+    - matplotlib
```
Comment (xhochy) [inline recipes/networkit/meta.yaml]:
```suggestion
    - matplotlib-base
```

File: recipes/networkit/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    - seaborn >=0.8.1
+    - ipython
+    - jupyter
```
Comment (xhochy) [inline recipes/networkit/meta.yaml]: Better use `jupyter_console` to also not require to install all Jupyter frontends.

```suggestion
    - jupyter_console
```

File: recipes/networkit/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    - '{{PYTHON}} -m pip install --ignore-installed -vv .'
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake >=3.5
+    - ninja >=1.9.0
+  host:
+    - python
```
Comment (xhochy) [inline recipes/networkit/meta.yaml]: Normally only `python` and `pip` are needed to install a package, all other can be omitted. Please check this.

Comment (lucventurini) [reply recipes/networkit/meta.yaml]: I was having problems in compiling from source without some of the packages (definitely sphinx, cython, openmp, wheel, ipython and jupyter are needed). I have implemented all other suggestions, thank you!

Comment (xhochy) [reply recipes/networkit/meta.yaml]: Good to know. This then probably needs some work upstream then.

Comment (lucventurini) [reply recipes/networkit/meta.yaml]: Thank you! I will communicate with the developers (I just took charge of making the recipe).

**Pull request merged**